File: extract_urls.py
import urllib.request
import re

url = "https://www.nebaministry.org/"
req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
try:
    with urllib.request.urlopen(req) as response:
        html = response.read().decode('utf-8')
        urls = re.findall(r'https?://[^\s"\'<>]+\.(?:jpg|jpeg|png|webp)', html, re.IGNORECASE)
        urls = list(set(urls))
        with open('urls.txt', 'w') as f:
            for u in urls:
                f.write(u + '\n')
        print(f"Saved {len(urls)} URLs")
except Exception as e:
    print(f"Error: {e}")

File: extract_urls.js
const https = require('https');
const fs = require('fs');

https.get('https://www.nebaministry.org/', (res) => {
  let data = '';
  res.on('data', chunk => data += chunk);
  res.on('end', () => {
    const urls = data.match(/https?:\/\/(?:[a-zA-Z0-9-]+\.)+[a-zA-Z]{2,}(?:\/?[^\s"'>\\]*\.(?:png|jpg|jpeg|webp|gif))/gi) || [];
    const uniqueUrls = [...new Set(urls)];
    fs.writeFileSync('urls.txt', uniqueUrls.join('\n'));
    console.log(`Saved ${uniqueUrls.length} URLs to urls.txt`);
  });
}).on('error', (e) => {
  console.error(e);
});
